mediacloud/mediawords/util/web/user_agent/throttled.py

```diff
@@ -0,0 +1,67 @@
+"""Implement ThrottledUserAgent as a sub class of mediawords.util.web.UserAgent with per domain throttling."""
+
+import typing
+
+import mediawords.db
+import mediawords.util.config
+from mediawords.util.web.user_agent import UserAgent
+from mediawords.util.web.user_agent.request.request import Request
+from mediawords.util.web.user_agent.response.response import Response
+
+from mediawords.util.log import create_logger
+log = create_logger(__name__)
+
+# default amount of time in between requests
+_DEFAULT_DOMAIN_TIMEOUT = 10
+
+
+class McThrottledUserAgentTimeoutException(Exception):
+    """Exception raised when a ThrottledUserAgent request fails to get a domain request lock."""
+
+    pass
+
+
+class ThrottledUserAgent(UserAgent):
+    """Add per domain throttling to mediawords.util.web.UserAgent."""
+
+    def __init__(self, db: mediawords.db.DatabaseHandler, domain_timeout: typing.Optional[int]=None) -> None:
```

Comment:
maybe `domain_timeout: int=_DEFAULT_DOMAIN_TIMEOUT`?

Reply:
I want the first fallback to be to go to mediawords.yml, and only after that to go to DEFAULT_DOMAIN_TIMEOUT.  I can't do that if I put the constant default in the parameter list.